src/w3c/templates/headers.js

```diff
@@ -2,16 +2,40 @@ import "deps/hyperhtml";
 import showLogo from "./show-logo";
 import showPeople from "./show-people";
 import showLink from "./show-link";
+import { pub } from "core/pubsubhub";
 
 export default conf => {
   const html = hyperHTML;
+  const specTitleElem =
+    document.querySelector("h1#title") || document.createElement("h1");
+  if (specTitleElem.parentElement) {
+    specTitleElem.remove();
+    conf.title = specTitleElem.textContent.trim();
+  } else {
+    if (document.title !== conf.title) {
+      pub("warn", "The document's title and the `<title>` element differ.");
+    }
+    specTitleElem.textContent = conf.title;
+    specTitleElem.id = "title";
+  }
+  specTitleElem.classList.add("title", "p-name");
+
+  const specSubTitleElem =
+    document.querySelector("h2#subtitle") || document.createElement("h2");
```

Comment:
You might want to add create a `h2` for subtitle only if `conf.subtitle` is given. 

Saying this if we need the same test conditions to hold for `"handles missing subtitle"` in https://github.com/w3c/respec/blob/9ae571a97d9162316c011c4946b1b16a205d23ef/tests/spec/w3c/headers-spec.js#L224-L233

Reply:
Nice suggestion. Though I guess now we can do away with the keeping the subtitle optional as we are anyways adding the subtitle element by my code. whatsay @marcoscaceres  ?

Reply:
If `conf.subtitle` is not specified and neither `h2#subtitle` is created by user, it creates an empty `<h2 id="subtitle"></h2>`. Might need to look into that too (although an empty `h2#subtitle` doesn't hurt)

Reply:
Prefer we don’t have empty elements, as they risk getting styled.

Reply:
So I should only include a `subTitleElement` if it exists in `conf.subTitle` ?

Reply:
Yep, exactly.

Reply:
ohk fine :+1: